File: java/Flyweight/src/flyweight/FlightWorldMap.java
package flyweight;

import java.util.Random;

public class FlightWorldMap {

	String flights[] = { "Delta", "American Airlines", "United", "BA", "Air France", "KLM" };
	Random random = new Random();
	FlightFactory flightFactory = new FlightFactory();
	
	private double getRandomLatitude() {
		double value = Math.random() * 90;
		return random.nextBoolean() ? value : value * -1D;
	}
	
	private double getRandomLongitude() {
		double value = Math.random() * 180;
		return random.nextBoolean() ? value : value * -1D;
	}
	
	private String getRandomFlight()
	{
		return flights[random.nextInt(5)];
	}
	
	
	private int getRandomFlightNo() {
		return random.nextInt(999);
	}
	
	public void displayAllFlights() {
		for(int i = 0; i < 100000; i++) {
			Flight flight = flightFactory.getFlight(getRandomFlight()); // >3
			flight.drawOnMap(getRandomFlightNo(), getRandomLatitude(), getRandomLongitude());
		}
	}
}

File: ruby/Flyweight/flight_world_map.rb
require_relative 'flight_factory'

class FlightWorldMap
  def initialize
    @flights = ["Delta", "American Airlines", "United", "BA", "Air France", "KLM" ]
    @ffactory = FlightFactory.new
  end

  def get_random_latitude
    rand(0..90)
  end

  def get_random_longitude
    rand(0..180)
  end

  def get_random_flight
    return @flights[rand(0..5)]
  end

  def get_random_flight_no
    rand(100..900)
  end

  def display_all_flight
    for i in 1..100000
      flight = @ffactory.get_flight(get_random_flight) # > 3
      flight.draw_on_map(get_random_flight_no(), get_random_latitude(), get_random_longitude());
    end
  end
end